### DISCUSSION TITLE
thanhle.blog/en/frontend-performance-pattern-en}

### DISCUSSION THREAD

**Discussion by giscus** [thanhlmm/blog#89] [Ideas]
# thanhle.blog/en/frontend-performance-pattern-en}

Pattern to optimize frontend performance

https://thanhle.blog/en/blog/frontend-performance-pattern-en

**Comment by trinhvanminh** [comment]
Hiện tại thì em đang gặp một số vấn đề với `Split code/Lazyload` -> **chunk failed to load error**

Ở đây em có tìm được một số bài nhưng chưa cảm thấy ổn lắm 
- https://mitchgavan.com/code-splitting-react-safely/
- https://www.codemzy.com/blog/fix-chunkloaderror-react

Hiện tại thì em đang dùng service worker để xử lý vấn đề trên nhưng mà nó có hai cách để handle
- Force refresh immediately
- Prompt refresh message and let user trigger action

Em đọc tiếp thì thấy bên dưới a có đề cập tới bài viết `update app in background` nên em muốn hỏi a là để **fix the chunk load error?** thì cách tốt nhất mình nên làm là gì ạ

**Reply by thanhlmm** [reply]
Thường thì người ta hay chọn là `Prompt refresh message and let user trigger action`

Còn cách giải quyết của mình là thay vì Prompt thì khi users change route sẽ tương đương với việc prompt đồng ý refresh. 
Nó sẽ mang lại trải nghiệm tốt hơn vì ko phải hỏi user, user có khi cũng chả biết nó là gì, mà khi route qua trang mới thì nó chỉ nháy nhẹ 1 cái

Hy vọng giúp dc bạn
